cli/Main.d.ts

```diff
@@ -1,15 +1,9 @@
-interface Renderer {
-    name: string;
-    extension: string;
-    aceMode: string;
-}
-
 type SourceCode = string;
 type ErrorMessage = string;
 
 interface Main {
     renderers: Renderer[];
-    main(config: Config): Either<ErrorMessage, SourceCode>;
+    main(options: {[name: string]: string}): ((config: Config) => Either<ErrorMessage, SourceCode>);
```

Comment:
Can you uncurry this? `Main.main` is always called from Type/JavaScript so we could just do the parameters as a record.

Reply:
Or could `options` be a property of `Config`? That seems better.